### PR TITLE
Re-write related repos readme template

### PR DESCRIPTION
**Pull request by lbussell**
I'm trying to remove lots of the hardcoded nonsense in this template. This should address #5596 as well when it's completed. I'm making sure at the same time that this template still has a sensible output for the main branch too.

TODO:

- [x] Make samples READMEs not reference nightly repos
- [x] Make README.md RelatedRepos reference non-nightly repos instead of nightly ones
- [x] Try to share implementation with Featured Repos

### CODE REVIEW

**Comment by lbussell** [comment]
I also ran these changes against the main branch. Here is a diff of that: https://github.com/lbussell/dotnet-docker/commit/e81003a82200095c1ab8a04e73976923f394a4dd

There are a few changes there, but it should be overall more consistent with the nightly branch.

**Review by mthalman — APPROVED** [review]

**Comment by lbussell** [comment]
I switched the ordering back around to have the samples at the bottom to match the main branch. Now, the diff on the main branch is only the fix for https://github.com/dotnet/dotnet-docker/issues/5596 and the addition of the nightly/monitor/base repo to the monitor/base readme.

File: README.md [hunk 1]
```diff
@@ -79,6 +79,12 @@ The [Image Variants documentation](https://github.com/dotnet/dotnet-docker/blob/
 .NET:
 
 * [dotnet](https://hub.docker.com/r/microsoft/dotnet/): .NET
+* [dotnet/sdk](https://hub.docker.com/r/microsoft/dotnet-sdk/): .NET SDK
+* [dotnet/aspnet](https://hub.docker.com/r/microsoft/dotnet-aspnet/): ASP.NET Core Runtime
+* [dotnet/runtime](https://hub.docker.com/r/microsoft/dotnet-runtime/): .NET Runtime
+* [dotnet/runtime-deps](https://hub.docker.com/r/microsoft/dotnet-runtime-deps/): .NET Runtime Dependencies
+* [dotnet/monitor](https://hub.docker.com/r/microsoft/dotnet-monitor/): .NET Monitor Tool
+* [dotnet/aspire-dashboard](https://hub.docker.com/r/microsoft/dotnet-aspire-dashboard/): .NET Aspire Dashboard
```
Comment (mthalman) [inline README.md:87]: Thinking about this more... why is it necessary to list all of these for the product family readme when they're already listed above in the featured section?

Comment (lbussell) [reply README.md:87]: These are the non-nightly versions. The featured section contains the nightly versions.

**Review by mthalman — APPROVED** [review]

**Pull request merged**